int32 x
int32 y
uint32 width
uint32 height
